### PR TITLE
MRC-1210 allow fhir server to not have searches be case sensitive

### PR DESCRIPTION
**Pull request by bwell-bruce**
Searches will now return partial and case-insensitive results.

### CODE REVIEW

**Comment by bwell-bruce** [comment]
r4.js is mostly contains white space (auto format) changes. Skip down to the comments where I have pointed out the changes.

File: src/operations/query/r4.js [hunk 1]
```diff
@@ -147,246 +151,277 @@ function convertGraphQLParameters(queryParameterValue, args, queryParameter) {
...
+              throw new Error('Unknown type=' + propertyObj.type);
+          }
+        } else if (args[`${queryParameter}:missing`]) {
+          // handle check for missing values
+          const exists_flag = !isTrue(args[`${queryParameter}:missing`]);
+          and_segments.push({
+            [`${propertyObj.field}`]: { $exists: exists_flag },
+          });
+          columns.add(`${propertyObj.field}`);
+        } else if (args[`${queryParameter}:contains`]) {
```
Comment (bwell-bruce) [inline src/operations/query/r4.js:390]: This is the main change to this file. The addition of the 'contains' modifier.

File: src/operations/query/r4.js [hunk 2]
```diff
@@ -147,246 +151,277 @@ function convertGraphQLParameters(queryParameterValue, args, queryParameter) {
...
+          and_segments.push({
+            [`${propertyObj.field}`]: { $regex: args[`${queryParameter}:contains`], $options: 'i' },
+          });
+          columns.add(`${propertyObj.field}`);
+        } else if (args[`${queryParameter}:above`]) {
+          // handle check for above the passed in  value
+          and_segments.push({
+            [`${propertyObj.field}`]: { $gte: args[`${queryParameter}:above`] },
```
Comment (bwell-bruce) [inline src/operations/query/r4.js:398]: Changing $gt to $gte will allow greater than and equals.

File: src/operations/query/r4.js [hunk 3]
```diff
@@ -147,246 +151,277 @@ function convertGraphQLParameters(queryParameterValue, args, queryParameter) {
...
+          });
+          columns.add(`${propertyObj.field}`);
+        } else if (args[`${queryParameter}:below`]) {
+          // handle check for below the passed in value
+          and_segments.push({
+            [`${propertyObj.field}`]: { $lte: args[`${queryParameter}:below`] },
```
Comment (bwell-bruce) [inline src/operations/query/r4.js:404]: Changing $lt to $lte will allow less than and equals.

File: src/operations/query/r4.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-    dateQueryBuilder,
-    referenceQueryBuilder,
-    // nameQueryBuilder,
-    // stringQueryBuilder,
-    // addressQueryBuilder,
-    tokenQueryBuilder,
-    dateQueryBuilderNative
+  dateQueryBuilder,
+  referenceQueryBuilder,
+  // nameQueryBuilder,
```
Comment (DarylCheung) [inline src/operations/query/r4.js:4]: Remove commented out variables

Comment (bwell-bruce) [reply src/operations/query/r4.js]: Removed

**Review by DarylCheung — APPROVED** [review]

**Pull request merged**